### DISCUSSION TITLE
Signing Transaction failing when using AppWallet?

### DISCUSSION THREAD

**Discussion by 0xMakka** [MeshJS/mesh#184] [General]
Hi guys, I'm pretty new the to the Cardano ecosystem and working on a new token. 

In Node.js, I'm using AppWallet passing along my mnemonic phrase,

```
const wallet = new AppWallet({
  networkId: 1,
  fetcher: blockchainProvider,
  submitter: blockchainProvider,
  key: {
    type: "mnemonic",
    words: [ ... ],
  },
});
```

I'm looking to send a token from the above wallet to another wallet, I'm building up the transaction and all this works fine, but it fails at the point of signing the transaction from the wallet. 

I'm receiving the following error `Error: An error occurred during signTx: Error: Missing key witness for: ...`. Has anyone came across this issue before, I would have assumed the wallet instance would contain the signing key?

```
    const tx = new Transaction({ initiator: wallet })
      .setRequiredSigners([address])
      .setTxInputs(selectedUserUtxos)
      .setChangeAddress(recipientAddress)
      .sendLovelace(address, "2000000")
      .setTxInputs(selectedWalletUtxos)
      .sendAssets(recipientAddress, [
        {
          unit: "...",
          quantity: "1",
        },
      ])
      .setChangeAddress(address);

    const unsignedTx = await tx.build();
    const signedTx = await wallet.signTx(unsignedTx);
    const txHash = await wallet.submitTx(signedTx);
```

**Comment by jinglescode** [comment]
@0xMakka use the latest [`MeshWallet`](https://meshjs.dev/apis/meshwallet) if youre not sure about how to use `AppWallet`
